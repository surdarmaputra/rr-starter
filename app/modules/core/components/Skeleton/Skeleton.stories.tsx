import type { Meta, StoryObj } from '@storybook/react';

import { Skeleton, type SkeletonProps } from './Skeleton';

type Story = StoryObj<typeof Skeleton>;

const meta: Meta = {
  component: Skeleton,
  title: 'Core/Components/Skeleton',
};

export default meta;

export const Default: Story = {
  render: (args: SkeletonProps) => (
    <div className="flex flex-col gap-2">
      <Skeleton {...args} className="h-32 w-32" />
      <Skeleton {...args} className="h-32 w-32 rounded-full" />
      <Skeleton {...args} className="h-4 w-32 rounded-full" />
    </div>
  ),
};
