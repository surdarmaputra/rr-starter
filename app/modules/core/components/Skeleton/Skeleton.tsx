import { cn } from '~/modules/core/libs/shadcn/utils';

export type SkeletonProps = React.HTMLAttributes<HTMLDivElement>;

function Skeleton({ className, ...props }: SkeletonProps) {
  return (
    <div
      className={cn(
        'animate-pulse rounded-lg bg-accent duration-1000',
        className,
      )}
      {...props}
    />
  );
}

export { Skeleton };
