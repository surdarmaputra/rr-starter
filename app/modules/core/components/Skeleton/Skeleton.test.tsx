import { render, screen } from '@testing-library/react';
import { describe, expect, test } from 'vitest';

import { Skeleton } from './Skeleton';

describe('Skeleton', () => {
  function renderComponent() {
    return render(
      <Skeleton className="h-4 w-12" data-testid="Skeleton-testid" />,
    );
  }

  test('renders with data-testid', async () => {
    renderComponent();
    await screen.findByTestId('Skeleton-testid');
  });

  test('renders snapshot', async () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });
});
