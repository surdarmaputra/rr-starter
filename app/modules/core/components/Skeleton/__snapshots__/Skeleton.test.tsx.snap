// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Skeleton > renders snapshot 1`] = `
<div>
  <div
    class="animate-pulse rounded-lg bg-accent duration-1000 h-4 w-12"
    data-testid="Skeleton-testid"
  />
</div>
`;
